geode-core/src/main/java/org/apache/geode/internal/cache/DiskInitFile.java

```diff
@@ -1310,7 +1311,16 @@ private void writeIFRecord(ByteBuffer bb) throws IOException {
   private void writeIFRecord(ByteBuffer bb, boolean doStats) throws IOException {
     assert lock.isHeldByCurrentThread();
     if (closed) {
-      throw new DiskAccessException("The disk store is closed", parent);
+      parent.getCache().getCancelCriterion().checkCancelInProgress();
+
+      if (parent.isClosed() || parent.isClosing()) {
+        throw new CacheClosedException("The disk store is closed or closing");
```

Comment:
I am not sure why `CacheClosedException` is thrown here. If the cache is closed, I assume `checkCancelInProgress` will throw `CacheClosedException`.

Reply:
The cache is not closed yet. But is going to be closed by the exception with disk-store. Also the cache close happens in the async thread, that will take some time to get initiated.

Reply:
Right. Looking into `cancelInProgress()` in `DiskStoreImpl.Stopper`, it only checks`isClosed()`. But it doesn't check `isClosing()`. It is good to have `if (parent.isClosed() || parent.isClosing())` here.

Reply:
I wonder if we should use OplogCancelledException instead of CacheClosedException here?